arangod/Aql/IResearchViewExecutor.cpp

```diff
@@ -706,33 +732,89 @@ bool IResearchViewExecutorBase<Impl, Traits>::writeRow(ReadContext& ctx,
     } else {
       UNUSED(bufferEntry);
     }
-    if constexpr (Traits::MaterializeType == MaterializeType::LateMaterializedWithVars) {
-      auto const& columnsFieldsRegs = infos().getOutNonMaterializedViewRegs();
-      TRI_ASSERT(!columnsFieldsRegs.empty());
-      auto const& storedValues = _indexReadBuffer.getStoredValue(bufferEntry);
-      int max = (--columnsFieldsRegs.end())->first;
-      TRI_ASSERT(max >= IResearchViewNode::SortColumnNumber);
-      if (IResearchViewNode::SortColumnNumber == max) {
-        ++max;
-      }
-      auto columnFieldsRegs = columnsFieldsRegs.cbegin();
-      if (IResearchViewNode::SortColumnNumber == columnFieldsRegs->first) {
-        if (ADB_UNLIKELY(!writeStoredValue(ctx, storedValues, max, columnFieldsRegs->second))) {
-          return false;
-        }
-        ++columnFieldsRegs;
+
+    return true;
+  }
+
+  return false;
+}
+
+template<typename Impl, typename Traits>
+void IResearchViewExecutorBase<Impl, Traits>::getStoredValue(irs::document const* doc, std::vector<irs::bytes_ref>& storedValue, int index,
```

Comment:
let's use `size_t` instead of `int` for `index` argument to eliminate useless type conversion